app/models/claim/claims_preventing_payment_finder.rb

```diff
@@ -0,0 +1,29 @@
+class Claim
+  class ClaimsPreventingPaymentFinder
+    attr_reader :claim
+
+    def initialize(claim)
+      @claim = claim
+    end
+
+    def claims_preventing_payment
+      @claims_preventing_payment ||= find_claims_preventing_payment
+    end
+
+    def find_claims_preventing_payment
```

Comment:
This could probably be a private method, as it doesn't get used outside the class

Reply:
Nice spot!